docs/source/conf.py

```diff
@@ -2151,6 +2151,7 @@
     "FileCheck",
     "FloatType",
     "FunctionSchema",
+    "GeneratorType",
```

Comment:
This needs to be documented, it is not ok to add anything to this list.

Reply:
@albanD Much like all of the other `*Type` classes, its a new non user facing type for which docs won't exist. I think its fine to add this class to the `coverage_ignore_classes`. Is there a particular reason why it can't?

Reply:
Because we have a strict policy that all public APIs must be documented.

There are quite a few historical ones that are not yet documented but we want to fix these to empty this list.

Reply:
Given that there are no other `*Type` classes documented anywhere, its unclear where such documentation should be added. The `GeneratorType` jit type has existed long before this PR. Its just only now part of the public API as there was seemingly never a need for it in the past. But now there is as TorchScript infers its types from the type annotation (see changes to `torch/jit/annotations.py`).

Seeing as none of the other `*Type` classes are documented anywhere either, I think this falls into the "historical" category. When all the jit `*Type` classes are documented, the `GeneratorType` can be documented at the same time.

Reply:
I'm fine with moving this type to be private (rename it to _GeneratorType) if we say that we don't want to document it right now.
Then we can move it to a public once we properly document it.

Reply:
done